src/main/java/ladder/domain/Ladder.java

```diff
@@ -0,0 +1,54 @@
+package ladder.domain;
+
+import ladder.domain.factory.PlayerFactory;
+import ladder.domain.vo.Rows;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class Ladder {
+
+    private final List<Rows> rowsList;
+
+    public Ladder(List<String> playersNames,
+                  Integer height){
+        this(create(PlayerFactory.create(playersNames), height));
+    }
+
+    public Ladder(Players players,
+                  Integer height){
+        this(create(players, height));
+    }
+
+    public Ladder(List<Rows> rowsList) {
+        this.rowsList = rowsList;
+    }
+
+    private static List<Rows> create(Players players,
+                                     Integer height){
+        if (players.count() < 2)
+            throw new IllegalArgumentException("사다리 생성에 필요한 참가자는 최소 2명 이상이어야 합니다.");
+        if (height < 2)
+            throw new IllegalArgumentException("사다리 높이는 최소 2 이상이어야 합니다.");
+        return IntStream.range(0, height)
+                .mapToObj(i -> new Rows(players.count() - 1))
+                .collect(Collectors.toList());
+    }
+
+    public List<Rows> rowsList(){
+        return Collections.unmodifiableList(rowsList);
+    }
+
+    public Integer height(){
+        return rowsList.size();
+    }
+
+    public Integer width(){
+        return rowsList().stream()
+                .findFirst()
+                .map(rows -> rows.rows().size())
+                .orElseThrow(() -> new IllegalStateException("알 수 없는 오류 입니다."));
```

Comment:
Stream을 굳이 써야 할 필요가 없을 것 같아요! 간단하게 구현해도 문제가 없을 것 같네요 ㅎㅎ

```suggestion
        if(rowsList.isEmpty()) {
            throw new IllegalStateException("사다리가 생성되지 않았습니다.");
        }
        
        return rowsList.get(0).size();
```

Reply:
한솔님이 말씀하신 내용도 이해 했습니다.
아무래도 rowsList.get(0) 이 좀 너무 날것 표현이지 아닌가 싶어 고민을 많이 하였습니다.
(물론 생성되면 다른 rows는 없을지라도 첫번째 rows는 존재하지만)
꼭 첫번째만 가져와야하는 이유가 없다고 생각이 되기도 하는데요... 🤔
우선 피드백을 수용하여 수정은 하였습니다.